web_stock/static/src/js/barcode.js

```diff
@@ -0,0 +1,192 @@
+/* Copyright 2016-2017 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+
+odoo.define('web_stock.barcode', function(require) {
+    'use strict';
+
+    var snippet_animation = require('web_editor.snippets.animation');
+    var ParentedMixin = require('web.mixins').ParentedMixin;
+    var BarcodeHandlerMixin = require('barcodes.BarcodeHandlerMixin');
+    var BarcodeParser = require('barcodes.BarcodeParser');
+    var Model = require('web.DataModel');
+    
+    var core = require('web.core');
+    var _t = core._t;
+
+    snippet_animation.registry.js_picking_form_barcode = snippet_animation.Class.extend(ParentedMixin, BarcodeHandlerMixin, {
+        
+        selector: '.js_picking_form_barcode',
+        loaded: false,
+        barcodeParser: false,
+        barcodeHistory: [],  // Most recent first
+        
+        // Let subclasses add custom behavior before onchange. Must return a deferred.
+        // Resolve the deferred with true proceed with the onchange, false to prevent it.
+        preOnchangeHook: function() {
+            return $.Deferred().resolve(this.loaded);
+        },
+    
+        start: function() {
+            this._super.apply(this, arguments);
+            // Compat w/ multi-barcodes update
+            this.el = this.$target[0];
+            this.actionMap = {
+                'product': this.handleProductScan,
+                'error': this.throwError,
+                'lot': this.handleLotScan,
+            };
+            if (!this.barcodeParser) {
+                var nomenclatureId = $('#barcodeNomenclatureId').val();
+                if (!nomenclatureId) {
+                    return;
+                }
+                this.barcodeParser = new BarcodeParser({
+                    'nomenclature_id': [nomenclatureId],
+                });
+                this.barcodeParser.load().then(this.onParserLoad);
+            }
+            this.$target.on('barcode_scanned', this.on_barcode_scanned);
+        },
+
+        onParserLoad: function() {
+            this.loaded = true;
+            console.log('Barcode parser initialized');
+        }
+
+        on_barcode_scanned: function(barcode) {
+            console.log(barcode);
+            var self = this;
+            // Call hook method possibly implemented by subclass
+            this.preOnchangeHook(barcode).then(function(proceed) {
```

Comment:
Hi @lasley I'm trying to test it, but I don't understand everything... I don't know why, I habe a picking, and product with barcode, I read this barcode and in this point proceed is false. Do you know that could be happing?
I also trying to migrate it to odoo 10.